### PR TITLE
feat(parser): support parsing placeholder

### PR DESCRIPTION
**Pull request by lewiszlw**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Support `?` placeholder.
Test not passed, could someone give it a fast review as I'm not very familiar about some detail code?
```
+---------- Input ----------
+select * from t where a = ?
+---------- Output ---------
+error:
+  --> SQL:1:27
+  |
+1 | select * from t where a = ?
+  | ------                -   ^ missing lhs or rhs for the binary operator
+  | |                     |
+  | |                     while parsing expression
+  | while parsing `SELECT ...`
+
```

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16030)

### CODE REVIEW

File: src/query/ast/src/ast/expr.rs [hunk 1]
```diff
@@ -227,6 +227,11 @@ pub enum Expr {
         span: Span,
         name: String,
     },
+    /// `?` or `$` Prepared statement arg placeholder
+    Placeholder {
+        span: Span,
+        name: String,
```
Comment (forsaken628) [inline src/query/ast/src/ast/expr.rs:233]: Consider adding an offset field, which would make the parameter binding easier to implement

FYI:
https://github.com/matrixorigin/matrixone/blob/bd8054941cfdab4cd94d17cd4260d11833a102b2/pkg/sql/parsers/dialect/mysql/mysql_lexer.go#L83-L86

https://github.com/matrixorigin/matrixone/pull/3957/files#diff-6f8d2e3805b41b9704c5d638c08078e8a7e5f6812ffa344ad607746319026e1fR856

**Comment by b41sh** [comment]
Hi @lewiszlw thanks for your contribution.
`?` is also used by `JsonOperator::Question`, and `?` is first matched by `json_op` function, so `placeholder` is never come into effect. We need add some extra work to check whether `?` is between two exprs to convert `JsonOperator::Question` to `Placeholder`.

**Comment by sundy-li** [comment]
Is is used to prepare stmt?  We don't have a plan to support prepare stmt

**Comment by lewiszlw** [comment]
Thanks for your help.
Yes, I added this on my fork for prepared stmt. Closing the pr as you don't have plan to support prepared stmt.

**Comment by forsaken628** [comment]
> Is is used to prepare stmt like mysql ? We don't have a plan to support prepare stmt in databend.
> 
> It could be supported on client side (JDBC...)

But here's an open issue #4620

**Comment by sundy-li** [comment]
The issue is outdated.  Sorry for the confusion.

**Pull request closed without merging**